gpkit/constraint_set.py

```diff
@@ -0,0 +1,145 @@
+import numpy as np
+from .small_classes import Numbers, HashVector, KeySet, KeyDict
+from .constraint_meta import LocallyApproximableConstraint, GPConstraint
+
+
+class ConstraintSet(LocallyApproximableConstraint, GPConstraint):
```

Comment:
I know I've already complained about this, but I still really don't like that this double inheritance is interpreted as an OR instead of an AND.

I see that a `ConstraintSet` contains things of those two types, but I don't see that it should inherit from (i.e. be a) `GPConstraint` -- since it might contain things that simply aren't GPcompatible except through local approximation.